storage/innobase/fsp/fsp0file.cc

```diff
@@ -346,10 +348,22 @@ Datafile::read_first_page(bool read_only_mode)
 		if (!fsp_flags_is_valid(m_flags, m_space_id)) {
 			ulint cflags = fsp_flags_convert_from_101(m_flags);
 			if (cflags == ULINT_UNDEFINED) {
-				ib::error()
-					<< "Invalid flags " << ib::hex(m_flags)
-					<< " in " << m_filepath;
-				return(DB_CORRUPTION);
+				switch (fsp_flags_is_incompatible_mysql(m_flags)) {
+				case 0:
+					ib::error()
+						<< "Invalid flags " << ib::hex(m_flags)
+						<< " in " << m_filepath;
+					return(DB_CORRUPTION);
+				case 0b11:
```

Comment:
In 10.3, we do not explicitly specify the C++ dialect, but older compilers default to C++98. Binary literals were only introduced in C++11, which is available starting with MariaDB Server 10.4.